snippets/fsharp/VS_Snippets_CLR_System/system.console.class/fs/example3.fs

```diff
@@ -0,0 +1,169 @@
+﻿
+// dump a range of Unicode characters as a 16x16 array
+// <Snippet4>
+open System
+open System.IO
+open System.Globalization
+open System.Text
+
+type uint = uint32
+
+let roundDownToMultipleOf (b:uint) (u:uint) : uint =
+    u - (u % b)
+
+let roundUpToMultipleOf (b:uint) (u:uint) : uint =
+    roundDownToMultipleOf b u |> (+) b
+
+let displayRange (start:uint) (``end``:uint) : unit =
+
+    let mutable upperRange : uint = 0x10FFFFu
+    let mutable surrogateStart : uint = 0xD800u
+    let mutable surrogateEnd : uint = 0xDFFFu
+   
+    let mutable start = start
+    let mutable ``end`` = ``end``
+    if ``end`` <= start
+    then
+        let t = start
+        start <- ``end``
+        ``end`` <- t
+
+    // Check whether the start or end range is outside of last plane.
+    if start > upperRange
+    then
+        raise (ArgumentException(String.Format("0x{0:X5} is outside the upper range of Unicode code points (0x{1:X5})", start, upperRange)))
+    if ``end`` > upperRange
+    then
+        raise (ArgumentException(String.Format("0x{0:X5} is outside the upper range of Unicode code points (0x{0:X5})",
+                                               ``end``, upperRange)))
+
+    // Since we're using 21-bit code points, we can't use U+D800 to U+DFFF.
+    if ((start < surrogateStart && ``end`` > surrogateStart) || (start >= surrogateStart && start <= surrogateEnd ))
+    then raise (ArgumentException(String.Format("0x{0:X5}-0x{1:X5} includes the surrogate pair range 0x{2:X5}-0x{3:X5}", 
+                                               start, ``end``, surrogateStart, surrogateEnd)))
+    let last = roundUpToMultipleOf 0x10u ``end``
+    let first = roundDownToMultipleOf 0x10u start
+
+    let rows = (last - first) / 0x10u
+
+    for r in 0u..(rows - 1u) do
+        // Display the row header.
+        Console.Write("{0:x5} ", first + 0x10u * uint32 r)
+
+        for c in 0u..(0x10u - 1u) do
+            let cur = (first + 0x10u * r + c);
+            if cur  < start then
+               Console.Write(" {0} ", Convert.ToChar(0x20))
```

Comment:
Prefer `printfn "%c" (Convert.ToChart(0x20))`

Reply:
Should I try to change all of the Console calls to the F# builtin function counterparts? I feel like at least for specific snippets that are showing how to exercise particular Console methods, the raw Console method should be used but perhaps printf, printfn, ... Should be used in other places? Perhaps a good idea for a fire refactoring PR to distill even more idiomatic F# samples?

Reply:
I think that all of the "normal" console calls can be `printfn`. When it's a fancier call, keeping it as `Console.Write` seems perfectly fine to me.